kognic-io/kognic/io/tools/scene_input.py

```diff
@@ -73,42 +74,61 @@ def _start_scene_jobs(
     return [resource.create(scene, dryrun=dryrun, feature_flags=feature_flags).input_uuid for scene in scenes]
 
 
+def _chunkify(origin_list: list, chunk_size: int) -> List[list]:
+    return [origin_list[idx:idx+chunk_size] for idx in range(0, len(origin_list), chunk_size)]
+
 def _wait_for_scene_jobs(
     client: KognicIOClient,
     scene_uuids: List[SceneUuid],
     wait_timeout: float,
     sleep_time: float
 ) -> Generator[CreationResult, None, None]:
-    def poll_input_status(su: str) -> CreationResult:
-        """Wait for scene processing"""
-        start_time = time.time()
-        while time.time() - start_time < wait_timeout:
-            responses = client.input.get_inputs_by_uuids(input_uuids=[su])
-
-            if not responses:
-                return CreationResult(scene_uuid=su, status=CreationStatus.CREATED)
-
-            response = responses[0]
-            if response.status in [InputStatus.Failed]:
-                return CreationResult(scene_uuid=su, status=CreationStatus.FAILED, error=CreationError(stage=CreationStage.SCENE, message=response.error_message))
-
-            # Check timeout
-            time.sleep(sleep_time)
-        return CreationResult(scene_uuid=su, status=CreationStatus.FAILED, error=CreationError(stage=CreationStage.SCENE, message='Scene creation timed out'))
-
-    # Wait for scene jobs to complete. Yield creation results as they are or the process times out
-    # Having some good wait strategy here will be important. Think about big vs small scenes and the number of scenes
-    with ThreadPoolExecutor(max_workers=5) as executor:
-        features = [executor.submit(poll_input_status, su) for su in scene_uuids]
-        for future in as_completed(features):
-            yield future.result()
+    remaining_scenes_uuids = [*scene_uuids]
+    start_time = time.time()
+    chunk_size = NORMAL_CHUNK_SIZE
+    while time.time() - start_time < wait_timeout:
```

Comment:
A very minor thing that you can address if you want to
```suggestion
    while time.time() - start_time < wait_timeout and len(remaining_scenes_uuids) > 0:
```